Libraries/LibWeb/Layout/TreeBuilder.cpp

```diff
@@ -317,6 +318,12 @@ void TreeBuilder::restructure_block_node_in_inline_parent(NodeWithStyleAndBoxMod
         before_wrapper = last_child;
     } else {
         before_wrapper = nearest_block_ancestor.create_anonymous_wrapper();
+
+        // CSS 2.2 9.2.1.1 creates anonymous block boxes, but 9.4.1 states inline-block creates a BFC.
+        // Set wrapper to inline-block to participate correctly in the IFC within the parent inline-block.
+        if (nearest_block_ancestor.display().is_inline_block()) {
+            static_cast<NodeWithStyle&>(*before_wrapper).mutable_computed_values().set_display(CSS::Display::from_short(CSS::Display::Short::InlineBlock));
+        }
```

Comment:
Both your test case and https://web.whatsapp.com/ look good without these `display` changes. Can you provide a test case where this change is necessary?

Also, we should then just have that change in `.create_anonymous_wrapper()` if it _is_ meaningful.

Reply:
My changes still fix a bug visible in the screenshots below. Can you make sure you are spoofing the user agent?

Before: <img width="1112" alt="Screenshot 2025-05-23 at 08 34 04" src="https://github.com/user-attachments/assets/e17740d2-6bf9-48c5-813e-5f21b80c587c" />

After: 
<img width="1112" alt="Screenshot 2025-05-23 at 08 53 11" src="https://github.com/user-attachments/assets/cc31fdd4-47c5-4080-80f3-71c71ca4828f" />

Reply:
Moved the change into `.create_anonymous_wrapper()`.

Reply:
> My changes still fix a bug visible in the screenshots below. Can you make sure you are spoofing the user agent?

I did spoof the UA, otherwise you can't even see that screen. My point was that the multiple changes you originally made did not all contribute to the new test passing. Now you've moved the change to `create_anonymous_wrapper()`, that's no longer relevant :)

Reply:
Oh, thanks for the clarification.

Moving the changes to `create_anonymous_wrapper()` made the PR really tidy, thanks.